saleor/graphql/schema_printer.py

```diff
@@ -0,0 +1,374 @@
+        print_description(field, "  ", not i) + "  " + print_input_value(name, field)
+        for i, (name, field) in enumerate(type_.fields.items())
+    ]
+    return print_description(type_) + f"input {type_.name}" + print_block(fields)
+
+
+def print_fields(type_: Union[GraphQLObjectType, GraphQLInterfaceType]) -> str:
+    fields = [
+        print_description(field, "  ", not i)
+        + f"  {name}"
+        + print_args(field.args, "  ")
+        + f": {field.type}"
+        + print_deprecated(field.deprecation_reason)
+        for i, (name, field) in enumerate(type_.fields.items())
+    ]
+    return print_block(fields)
+
+
+def print_block(items: List[str]) -> str:
+    return " {\n" + "\n".join(items) + "\n}" if items else ""
+
+
+def print_args(args: Dict[str, GraphQLArgument], indentation: str = "") -> str:
+    if not args:
+        return ""
+
+    # If every arg does not have a description, print them on one line.
+    if not any(arg.description for arg in args.values()):
+        return (
+            "("
+            + ", ".join(print_input_value(name, arg) for name, arg in args.items())
+            + ")"
+        )
+
+    return (
+        "(\n"
+        + "\n".join(
+            print_description(arg, f"  {indentation}", not i)
+            + f"  {indentation}"
+            + print_input_value(name, arg)
+            for i, (name, arg) in enumerate(args.items())
+        )
+        + f"\n{indentation})"
+    )
+
+
+def print_input_value(name: str, arg: GraphQLArgument) -> str:
+    default_ast = ast_from_value(arg.default_value, arg.type)
+    arg_decl = f"{name}: {arg.type}"
+    if default_ast:
+        arg_decl += f" = {print_ast(default_ast)}"
+    return arg_decl
+
+
+def print_directive(directive: GraphQLDirective) -> str:
+    return (
+        print_description(directive)
+        + f"directive @{directive.name}"
+        + print_args(directive.args)
+        + " on "
+        + " | ".join(location for location in directive.locations)
+    )
+
+
+def print_deprecated(reason: Optional[str]) -> str:
+    if reason is None:
+        return ""
+    if reason != DEFAULT_DEPRECATION_REASON:
+        ast_value = print_ast(ast_from_value(reason))
+        return f" @deprecated(reason: {ast_value})"
+    return " @deprecated"
+
+
+def is_printable_as_block_string(value: str) -> bool:
+    """Check whether the given string is printable as a block string."""
+    if not isinstance(value, str):
+        value = str(value)  # resolve lazy string proxy object
+
+    if not value:
+        return True  # emtpy string is printable
+
+    is_empty_line = True
+    has_indent = False
+    has_common_indent = True
+    seen_non_empty_line = False
+
+    for c in value:
+        if c == "\n":
+            if is_empty_line and not seen_non_empty_line:
+                return False  # has leading new line
+            seen_non_empty_line = True
+            is_empty_line = True
+            has_indent = False
+        elif c in " \t":
```

Comment:
Wouldn't be better to create two statements?
```suggestion
        elif c == "\t":
            has_indent = True
        elif c == " ":
            has_indent = False
```
I am wondering why `has_indent` depends on `is_empty_line` ?

Reply:
This is code from `graphql-core`, not sure we want to spend time improving it. This entire file wile go away once we are able to upgrade.